ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h

```diff
@@ -13,6 +13,7 @@
 #  include <future>
 #endif
 
+#include <functional>
```

Comment:
Wondering why this happens only to clang16 and libc++.

Reply:
`std::function` is in `<functional>` (<https://en.cppreference.com/w/cpp/utility/functional/function>).But it may be included by other headers.